src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
```

Comment:
Hum, again be aware of micro benchmarks. If you replace `res = []` by `res = new Array(LEN)` (pre-allocating a dense array), you get from `~5000ms` to `~1700ms`, and the difference between a cache length becomes unoticable:

```
cached loop---
elapsed:  1646
op/s  607533414.3377886
non-cached loop---
elapsed:  1653
op/s  604960677.5559589
```